src/config/config.rs

```diff
@@ -69,18 +69,19 @@ pub struct GirVersion {
 impl GirVersion {
     fn new(gir_dir: impl AsRef<Path>) -> Self {
         let gir_dir = normalize_path(gir_dir);
+        let is_submodule = toplevel(&gir_dir) != Path::new(".").canonicalize().ok();
         Self {
             file_name: gir_dir
                 .file_name()
                 .map(|s| s.to_str().expect("OsStr::to_str failed").to_owned()),
-            hash: repo_hash(&gir_dir),
-            url: repo_remote_url(&gir_dir),
+            hash: is_submodule.then(|| repo_hash(&gir_dir)).flatten(),
```

Comment:
Theoretically we should add back `unwrap_or("???")` here if this is a submodule but we failed to extract the hash.